dcm4che-net-audit/src/main/java/org/dcm4che3/net/audit/AuditLogger.java

```diff
@@ -1518,7 +1521,7 @@ private void startIdleTimer() {
                 try {
                     Device device = conn.getDevice();
                     if (device.getScheduledExecutor() == null) {
-                        device.setScheduledExecutor(Executors.newSingleThreadScheduledExecutor());
+                        device.setScheduledExecutor(scheduledExecutor);
                     }
                     idleTimer = device.schedule(
                             new Runnable() {
```

Comment:
Should we also add a call to scheduledExecutor.shutdown() inside the onIdleTimerExpired() method below to clean up the Executor?

Reply:
No - then the thread pool is shutdown and can't be re-used by the next call which would use the idle timer